### ISSUE TITLE
File_Lock() at util.c:1043 might be called recursively

### ISSUE THREAD

**Issue by TAnhMinh** [quadrantsec/sagan#66]
The function File_lock is used in Sagan_Log at util.c:254. However, if we cannot obtain the lock inside File_Lock(), we will attempt to log this new error ("Unable to obtain LOCK on file...") which means calling Sagan_Log()  which calls File_lock() and so on, recursively. In the end, Sagan will crash. (This happened when we accidentally ran two instances of Sagan at once.)

Maybe the error "Unable to obtain LOCK on file..." should not be **logged** when File_Lock is called by Sagan_Log()? Instead, Sagan can print this error and end.

**Comment by quadrantsec** [comment]
Thanks for reporting this...

This makes a lot of sense and would certainly cause issues.  I am currently on vacation but will look at this next week,

**Comment by quadrantsec** [comment]
I had a chance to look at this on the way home and you are correct.  It's being called recursively.  

I've elected to remove File_Lock() / File_Unlock() from Sagan_Log().   File_Lock() / File_Unlock() is used when Sagan is making IPC calls to prevent two _**separate processes**_  (not threads) from over-writing data in /dev/shm ( mmap() ).  I think that I added it to logging "in case" two **_separate processes_**  are writing to the same "sagan.log" file.  This is of silly, IMHO.   

When running multiple Sagan processes on the same box, you should really have each instance pointing to its own "sagan.log" file.    

I also verified that fprintf() does atomic writes when threads are from the same process. 

Long story short,  I'm going to remove File_Lock() / File_Unlock() from Sagan_Log().    

Thank you for reporting this and let me know if you run into anything else.   If you see issues with this fix,  feel free to re-open this.
